client/components/date-range/header.js

```diff
@@ -0,0 +1,49 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { noop } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import Button from 'components/button';
+
+class DateRangeHeader extends Component {
+	static propTypes = {
+		onApplyClick: PropTypes.func,
+		onCancelClick: PropTypes.func,
+		applyButtonText: PropTypes.string,
+		cancelButtonText: PropTypes.string,
+	};
+
+	static defaultProps = {
+		onApplyClick: noop,
+		onCancelClick: noop,
+		applyButtonText: 'Apply',
```

Comment:
There seems to be an issue with translations here and it's rather easy to miss:

`DateRangeHeader` accepts it's props through `DateRange`, so in order for `applyButtonText` and `cancelButtonText` to be translated, the dev implementing `DateRange` would have to be sure to do something like so:

```jsx
<DateRange
  applyButtonText={ translate( 'Apply' ) }
  closeButtonText={ translate( 'Close' ) }
  ...
/>
```

Failing to do so would mean falling back to these untranslated defaults. The trouble is here that we're quite quiet about this requirement in that A) these props are not marked as required in this component & B) They're not advertised at all in `DateRange`, which is their real entry point.

I see a couple of options here...
We could keep these props as optional, but we take care of translating the default values, like so:

```jsx
class DateRangeHeader extends Component {
  ...

  static defaultProps = {
    onApplyClick: noop,
    onCancelClick: noop,
  };

  ...

  render() {
    const { translate } = this.props;
    return (
      <div className="date-range__popover-header">
        <Button>
          { this.props.cancelButtonText || translate( 'Cancel' ) }
        </Button>
        ...
      </div>
    );
  }
}

export default localize( DateRangeHeader );
```

Or we could remove the defaults and mark these props as required in both the `DateRangeHeader` and `DateRange` components.
I'd personally go for the first pattern as it alleviates the extra need to define these labels, yet still handles translations.

Which ever we go for, lets apply the same treatment to the other subcomponents.

BTW, I'm happy to scratch this up as work for another PR so as not to block this current work.

Reply:
Yep I'm going to bank that pattern. It's the right approach. Can't rely too much on `defaultProps`.

Fixed